.codegen/model.go.tmpl

```diff
@@ -18,20 +18,74 @@ import (
 	"github.com/databricks/databricks-sdk-go/marshal"
     "github.com/hashicorp/terraform-plugin-framework/types"
 )
+{{- $excluded := dict "ShareInfo"        (list "CreatedAt" "CreatedBy" "UpdatedAt" "UpdatedBy")
+                      "SharedDataObject" (list "AddedAt" "AddedBy" "Status") -}}
 {{range .Types}}
 {{- if or .Fields .IsEmpty}}
 {{.Comment "// " 80}}
 type {{.PascalName}} struct {
+    {{- $excluded := getOrDefault $excluded .PascalName (list) -}}
     {{- range .Fields}}
     {{.Comment "    // " 80}}
-    {{.PascalName}} {{template "type" .Entity}} `{{template "field-tag" . }}`{{end}}
+	{{- $data := dict "field" . "excluded" $excluded }}
+	{{template "field" $data}}{{if and .Entity.IsComputed (not (in $excluded .PascalName))}}{{ $data := dict "field" . "excluded" $excluded "effective" true }}{{printf "\n"}}{{template "field" $data}}{{end}}{{end}}
+}
+
+func (newState *{{.PascalName}}) SyncEffectiveFieldsDuringCreateOrUpdate(plan {{.PascalName}}) {
+  {{- range .Fields -}}
+  {{- if and .Entity.IsComputed (or .Entity.IsString .Entity.IsBool .Entity.IsInt64 .Entity.IsFloat64 .Entity.IsInt .Entity.Enum) -}}
+  {{- if not (in $excluded .PascalName) -}}
+  newState.Effective_{{.PascalName}} = newState.{{.PascalName}}
+  newState.{{.PascalName}} = plan.{{.PascalName}}
+  {{end}}
+  {{end}}
+  {{end}}
+}
+
+func (newState *{{.PascalName}}) SyncEffectiveFieldsDuringRead(existingState {{.PascalName}}) {
+  {{- range .Fields -}}
+  {{- if and .Entity.IsComputed (or .Entity.IsString .Entity.IsBool .Entity.IsInt64 .Entity.IsFloat64 .Entity.IsInt .Entity.Enum) -}}
+  {{- if not (in $excluded .PascalName) -}}
+  {{- $type := "" -}}
+  {{- if .Entity.IsString}}{{$type = "String"}}{{end}}
+  {{- if .Entity.IsBool}}{{$type = "Bool"}}{{end}}
+  {{- if .Entity.IsInt64}}{{$type = "Int64"}}{{end}}
+  {{- if .Entity.IsFloat64}}{{$type = "Float64"}}{{end}}
+  {{- if .Entity.IsInt}}{{$type = "Int64"}}{{end}}
+  {{- if .Entity.Enum}}{{$type = "String"}}{{end}}
+  if existingState.Effective_{{.PascalName}}.Value{{$type}}() == newState.{{.PascalName}}.Value{{$type}}() {
```

Comment:
Note: .Value<type> returns zero value (eg: "" string if string is null or unknown), is there a case where we would want to differentiate between these? Right now if one is null / one is unknown, then we will compare them to be the same.